mitosheet/src/jupyter/api.tsx

```diff
@@ -1279,15 +1279,19 @@ export default class MitoAPI {
     */
     async updateReplayAnalysis(
         analysisName: string,
-        stepImportDataListToOverwrite?: StepImportData[]
+        analysis?: SavedAnalysis,
+        stepImportDataListToOverwrite?: StepImportData[],
+        ignoreAuthorHash?: boolean
     ): Promise<MitoError | undefined> {
 
         const result: MitoError | undefined = await this.send({
             'event': 'update_event',
             'type': 'replay_analysis_update',
             'params': {
                 'analysis_name': analysisName,
-                'step_import_data_list_to_overwrite': stepImportDataListToOverwrite === undefined ? [] : stepImportDataListToOverwrite
+                'analysis': analysis === undefined ? null : analysis,
+                'step_import_data_list_to_overwrite': stepImportDataListToOverwrite === undefined ? [] : stepImportDataListToOverwrite,
+                'ignore_author_hash': ignoreAuthorHash === undefined ? false : true
```

Comment:
Hmm instead of having this one function do so many different things with different configurations, I think it might be better if we had the user press `Trust Analysis`, then we called an update function like `trust_analysis_update` which updated the author hash. Then if the analysis failed to update because of the imports / invalid steps, then we open up the PreReplayUpdateImportTaskpane which handles things as we normally do. 

Wrapping the security model into these different cases feels more fragile to me

Reply:
This was the first approach I took. The problem with this approach is that it leads to an extra roundtrip anytime you're replaying anything, which feels slow. There's also a lot of extra code you have to add that was pretty much just replicating what already exists. 

Let's see how this evolves. As of now, the backend has a single if statement, so it's not too messy currently. But I agree this is something to keep an eye on!